Panic in TestPreparingProgress

## Flaky Test

### Which jobs are failing

TestPreparingProgress

### CI link

https://github.com/tikv/pd/actions/runs/16645458912/job/47104946875?pr=9606

### Reason for failure (if possible)

```shell
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x44c0365]

goroutine 460356 [running]:
github.com/tikv/pd/tests/server/api.TestPreparingProgress.func7()
	/home/runner/work/pd/pd/tests/server/api/api_test.go:1189 +0xa5
github.com/stretchr/testify/assert.Eventually.func1()
	/home/runner/go/pkg/mod/github.com/stretchr/testify@v1.9.0/assert/assertions.go:1902 +0x34
created by github.com/stretchr/testify/assert.Eventually in goroutine 254
	/home/runner/go/pkg/mod/github.com/stretchr/testify@v1.9.0/assert/assertions.go:1902 +0x3d6
run all tasks takes 4m49.115057088s
make: *** [Makefile:266: ci-test-job] Error 1
```

--- a/tests/server/api/api_test.go
+++ b/tests/server/api/api_test.go
@@ -1141,6 +1141,11 @@ func TestPreparingProgress(t *testing.T) {
 		tests.MustPutRegion(re, cluster, uint64(i+1), uint64(i)%3+1, []byte(fmt.Sprintf("%20d", i)), []byte(fmt.Sprintf("%20d", i+1)), core.SetApproximateSize(10))
 	}
 	testutil.Eventually(re, func() bool {
+		if leader == nil {
+			re.NotEmpty(cluster.WaitLeader())
+			leader = cluster.GetLeaderServer()
+			return false
+		}
 		return leader.GetRaftCluster().GetTotalRegionCount() == core.InitClusterRegionThreshold
 	})
 
@@ -1159,8 +1164,16 @@ func TestPreparingProgress(t *testing.T) {
 		tests.MustPutStore(re, cluster, store)
 	}
 
+	re.NotEmpty(cluster.WaitLeader())
+	leader = cluster.GetLeaderServer()
 	if !leader.GetRaftCluster().IsPrepared() {
 		testutil.Eventually(re, func() bool {
+			leader = cluster.GetLeaderServer()
+			if leader == nil {
+				re.NotEmpty(cluster.WaitLeader())
+				leader = cluster.GetLeaderServer()
+				return false
+			}
 			if leader.GetRaftCluster().IsPrepared() {
 				return true
 			}
@@ -1177,6 +1190,12 @@ func TestPreparingProgress(t *testing.T) {
 	var p api.Progress
 	testutil.Eventually(re, func() bool {
 		defer triggerCheckStores()
+		leader = cluster.GetLeaderServer()
+		if leader == nil {
+			re.NotEmpty(cluster.WaitLeader())
+			leader = cluster.GetLeaderServer()
+			return false
+		}
 		// wait for cluster prepare
 		if !leader.GetRaftCluster().IsPrepared() {
 			leader.GetRaftCluster().SetPrepared()

